test/clientWithAutoConfirmEnabled.test.ts

```diff
@@ -61,6 +63,23 @@ test('signUp() the same user twice should throw an error', async () => {
   expect(user).toBeNull()
 })
 
+
+test('setUser() set the Auth headers on a new client', async () => {
+  expect(access_token).toBeTruthy()
+
+  const newClient = new GoTrueClient({
+    url: GOTRUE_URL,
+    autoRefreshToken: false,
+    persistSession: false,
+  })
+
+  newClient.setUser(access_token!)
+
+  const authBearer = newClient.session()?.access_token
+  expect(authBearer).toEqual(access_token)
+})
```

Comment:
here I 

1. create  a fresh client
2. set it to the previous user's jwt
3. check that `newClient.session()?.access_token` returns the correct jwt